### PR TITLE
Move Brave News Subscribe button to the left of the Tor button

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/26138

Before:
image

After:
image

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. With the Brave News Subscribe button feature enabled,
2. Navigate to https://theguardian.com
3. The subscribe button should be on the left of the Tor button

### CODE REVIEW

File: browser/ui/views/location_bar/brave_location_bar_view.h [hunk 1]
```diff
@@ -75,6 +76,7 @@ class BraveLocationBarView : public LocationBarView {
   friend class ::BraveActionsContainerTest;
   friend class ::RewardsBrowserTest;
   BraveActionsContainer* brave_actions_ = nullptr;
+  BraveNewsLocationView* brave_news_location_view = nullptr;
```
Comment (fallaciousreasoning) [inline browser/ui/views/location_bar/brave_location_bar_view.h:79]: Should I migrate all these to use raw_ptr?

File: components/brave_today/browser/publishers_controller.cc [hunk 2]
```diff
@@ -54,10 +54,13 @@ const mojom::Publisher* PublishersController::GetPublisherForSite(
...
 
-  const auto& site_origin = url::Origin::Create(site_url);
+  const auto& site_host = site_url.host();
   for (const auto& kv : publishers_) {
-    const auto& publisher_origin = url::Origin::Create(kv.second->site_url);
-    if (site_origin.IsSameOriginWith(publisher_origin)) {
+    const auto& publisher_host = kv.second->site_url.host();
+    // When https://github.com/brave/brave-browser/issues/26092 is fixed, this
+    // hack can be removed.
+    const auto& publisher_host_www = "www." + publisher_host;
```
Comment (fallaciousreasoning) [inline components/brave_today/browser/publishers_controller.cc:62]: This is the same check we're doing for source suggestions
https://github.com/brave/brave-core/blob/master/components/brave_today/browser/suggestions_controller.cc#L91

Hopefully the back end fix we need should be deployed within the next week or so.

File: browser/ui/views/location_bar/brave_location_bar_view.h (hunk 1 above)
Comment (simonhong) [reply browser/ui/views/location_bar/brave_location_bar_view.h:79]: `brave_news_location_view_`

File: browser/ui/views/location_bar/brave_news_location_view.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ public:
+  BraveNewsLocationView(Profile* profile, TabStripModel* tab_strip_model);
+  BraveNewsLocationView(const BraveNewsLocationView&) = delete;
+  BraveNewsLocationView& operator=(const BraveNewsLocationView&) = delete;
+  ~BraveNewsLocationView() override;
+
+  void Update();
+
+ private:
+  views::View* button_;
```
Comment (simonhong) [inline browser/ui/views/location_bar/brave_news_location_view.h:24]: `raw_ptr` :)

File: browser/ui/views/brave_actions/brave_actions_container.cc [hunk 4]
```diff
@@ -11,13 +11,13 @@
 
 #include "base/command_line.h"
 #include "base/feature_list.h"
+#include "base/one_shot_event.h"
```
Comment (simonhong) [inline browser/ui/views/brave_actions/brave_actions_container.cc:14]: nit: seems not related including with this PR?

**Review by simonhong — APPROVED** [review]
++ with some trivial comments 👍🏼

File: browser/ui/views/location_bar/brave_news_location_view.cc [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+#include "ui/views/layout/flex_layout_view.h"
+#include "ui/views/layout/layout_types.h"
+#include "ui/views/view.h"
+
+namespace {
+constexpr SkColor kSubscribedLightColor = SkColorSetRGB(76, 84, 210);
+constexpr SkColor kSubscribedDarkColor = SkColorSetRGB(115, 122, 222);
+}  // namespace
+
+class BraveNewsButtonView : public views::LabelButton,
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply browser/ui/views/location_bar/brave_news_location_view.cc:44]: Done!

File: browser/ui/views/location_bar/brave_news_location_view.h (hunk 3 above)
Comment (fallaciousreasoning) [reply browser/ui/views/location_bar/brave_news_location_view.h:24]: Removed :smile:

File: browser/ui/views/location_bar/brave_news_location_view.cc [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+    bubble_widget_ =
+        BraveNewsBubbleView::Show(this, tab_strip_->GetActiveWebContents());
+  }
+
+  SkColor GetIconColor(bool subscribed) const {
+    if (!subscribed)
+      return GetCurrentTextColor();
+
+    return dark_mode::GetActiveBraveDarkModeType() ==
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply browser/ui/views/location_bar/brave_news_location_view.cc:187]: Agreed, it'd be good to change this

**Comment by fallaciousreasoning** [comment]
@bsclifton for Brave Rewards client review (I had to touch a test)

**Review by bsclifton — APPROVED** [review]
rewards test change LGTM! ++

**Comment by stephendonner** [comment]
## Verified `PASSED` using 



Brave | 1.47.36 Chromium: 107.0.5304.91 (Official Build) nightly (x86_64)
-- | --
Revision | 3d5948960d62418160796d5831a4d2d7d6c90fa8-refs/branch-heads/5304@{#1097}
OS | macOS Version 11.7.1 (Build 20G918)

#### Confirmed the `Manage Subscriptions` button is to the left of the `Tor` button 

(Original scope also had it to the left of the `Share this page` button, but that would've required an additional patch to our Chromium codebase, as an override, so we dropped it.)

`Light` | `Dark`
--------|-------
Screen Shot 2022-11-01 at 3 26 32 PM | Screen Shot 2022-11-01 at 3 26 59 PM
